src/coreclr/jit/importer.cpp

```diff
@@ -3791,10 +3791,20 @@ void Compiler::impImportAndPushBox(CORINFO_RESOLVED_TOKEN* pResolvedToken)
         // and the other you get
         //    *(temp+4) = expr
 
-        if (opts.OptimizationDisabled())
+        // For minopts/debug code, try and minimize the total number
+        // of box temps by reusing an existing temp when possible.
+        bool useSharedBoxTemp = opts.OptimizationDisabled();
+
+        // However, when we're allowed to perform quick opts we want to still have exact classes for boxed enums
```

Comment:
Do we need to have a limit on how many of the non-shared temps we can have with light opts?

Reply:
It mitigated a quite big size regression in the coreclr_tests.run collection especially on arm64 - up to (`+800kb`) now all diffs are negative https://dev.azure.com/dnceng-public/public/_build/results?buildId=60597&view=ms.vss-build-web.run-extensions-tab

Reply:
Consider cases like a method with 1000 `HasFlag` calls. It will get extra 1000 temps that won't be optimized with light opts and end up consuming a lot more stack space.

Reply:
Ah, sorry, I thought you asked why we need to limit to enums this. Yes, it makes sense to have a hard limit of non-shared locals here, will play with spmi to find a good number.

Reply:
A while back I was looking at building per-type caches for temps like these (and for struct temps which cause even worse frame bloat) since you are much less likely to see huge numbers of different typed temps then huge numbers of different temps.

But since there tracking and invalidation logic and such to get right, it's not clear it is worth the trouble.